featuretools/entityset/entityset.py

```diff
@@ -155,6 +162,21 @@ def __getitem__(self, dataframe_name):
         name = self.id or "entity set"
         raise KeyError('DataFrame %s does not exist in %s' % (dataframe_name, name))
 
+    def __deepcopy__(self, memo):
+        cls = self.__class__
+        result = cls.__new__(cls)
+        memo[id(self)] = result
+        for k, v in self.__dict__.items():
+            copied_attr = copy.deepcopy(v, memo)
+            # --> figure out how to handle data description....
```

Comment:
Similar to the equality check, I wasn't sure if we might want to handle data description in a special way for copying.

Reply:
I'm thinking we don't need to handle this

Reply:
So we should just copy whatever is in the data description of the original entityset? Or not copy it and have `_data_description` on the new entityset be `None`?

Reply:
Ah, good question. I guess I was thinking it didn't need any special treatment, so I guess that would mean whatever is in the original entityset data description would get copied over.

Reply:
that makes sense to me!